### PR TITLE
Fix pkg/kubelet static checks

### PR DESCRIPTION
**Pull request by auxten**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

fixes `pkg/kubelet` static checks for
- S1039 unnecessary use of `fmt.Sprintf`
- S1008 simplify bool returns
- S1002 omit compares to bool
- fix capitalized error message

**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

File: pkg/kubelet/cm/topologymanager/policy_single_numa_node.go [hunk 1]
```diff
@@ -49,10 +49,10 @@ func filterSingleNumaHints(allResourcesHints [][]TopologyHint) [][]TopologyHint
 	for _, oneResourceHints := range allResourcesHints {
```
Comment (draveness) [inline pkg/kubelet/cm/topologymanager/policy_single_numa_node.go:49]: I think you should remove the package in https://github.com/kubernetes/kubernetes/blob/master/hack/.staticcheck_failures

Comment (auxten) [reply pkg/kubelet/cm/topologymanager/policy_single_numa_node.go:49]: I didn't see `kubelet` related pkg there

Comment (auxten) [reply pkg/kubelet/cm/topologymanager/policy_single_numa_node.go:49]: Maybe, the `.staticcheck_failures` needs an update. @BenTheElder @nolancon @sjenning

Comment (BenTheElder) [reply pkg/kubelet/cm/topologymanager/policy_single_numa_node.go:49]: this file is automatically kept up to date, PRs won't merge if it's up to date.

are you running staticcheck using the rule in the repo or with your own install? we use a specific version in CI

**Comment by SergeyKanzhelev** [comment]
there is a valid question from @BenTheElder

> this file is automatically kept up to date, PRs won't merge if it's not** up to date.
> 
> are you running staticcheck using the rule in the repo or with your own install? we use a specific version in CI

fixing these warnings is making code better. However there seems to be no mechanism in place enforcing these rules. So code may degrade back. Since it's just a general code cleanup, it's lgtm

/lgtm
/priority backlog

**Comment by auxten** [comment]
> there is a valid question from @BenTheElder
> 
> > this file is automatically kept up to date, PRs won't merge if it's not** up to date.
> > are you running staticcheck using the rule in the repo or with your own install? we use a specific version in CI
> 
> fixing these warnings is making code better. However there seems to be no mechanism in place enforcing these rules. So code may degrade back. Since it's just a general code cleanup, it's lgtm
> 
> /lgtm
> /priority backlog

Maybe, we should enforce strict staticcheck on every modified file on every PR.

**Pull request merged**